src/azure-cli-core/azure/cli/core/tests/test_argcomplete.py

```diff
@@ -0,0 +1,66 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+from unittest import mock, TestCase
+
+from azure.cli.core.mock import DummyCli
+from azure.cli.core.util import run_cmd
+
+
+class ArgcompleteScenarioTest(TestCase):
+
+    def argcomplete_env(self, comp_line, comp_point):
+        return {
+            'COMP_LINE': comp_line,
+            'COMP_POINT': comp_point,
+            '_ARGCOMPLETE': '1',
+            '_ARGCOMPLETE_IFS': ' ',
+            '_ARGCOMPLETE_SUPPRESS_SPACE': '0',
+            'ARGCOMPLETE_USE_TEMPFILES': '1',
+            '_ARGCOMPLETE_STDOUT_FILENAME': './argcomplete.out', }
+
+    def test_completion(self):
+        import os
+        import sys
+
+        from azure.cli.core.parser import AzCompletionFinder
+
+        if sys.platform == 'win32':
+            self.skipTest('Skip argcomplete test on Windows')
+
+        run_cmd(['az'], env=self.argcomplete_env('az account sh', '13'))
```

Comment:
Can this be changed to a `cli.invoke`?

Reply:
Yes, but we need to mock the `AzCompletionFinder` and `env`, like we did in line 59.
`run_cmd` is identical to the external autocompletion call.